data/is_ocaml_yet/web.md

```diff
@@ -0,0 +1,258 @@
+  - name: Browser
+    status: 🟢
+    description: |
+      OCaml has excellent support for running in the browser!
+      The [Js_of_ocaml](/p/js_of_ocaml/latest) compiler, which translates OCaml into JavaScript, is a well-established tool dating back to 2010. It integrates seamlessly with the existing OCaml ecosystem and powers numerous production applications.
+
+      In addition, [Melange](/p/melange/latest) is a more recent entry to the OCaml-JavaScript compiler space. Originating as a fork of [ReScript](https://rescript-lang.org/), Melange may not have the extensive track record of Js_of_ocaml, but it builds on a very strong foundation and already integrates well with Dune, the OCaml build system.
+    packages:
+    - js_of_ocaml
+    - melange
+    - brr
+    - gen_js_api
+  - name: Compression
+    status: 🟡
+    description: |
+      n/a
+    packages:
+    - zlib
+    - decompress
+    - camlzip
+    - lz4
+    - ezgzip
+    - hpack
+    - brotli
+    - zstd
+    - snappy
+    - lzo
+    - zstandard
+  - name: Crypto
+    status: 🟡
+    description: |
+      n/a
+    packages:
+    - cryptokit
+    - mirage-crypto
+    - nocrypto
+    - pkcs11
+    - tls
+    - hacl_x25519
+    - x509
+    - fiat-p256
+    - sha
+    - blake2
+    - blake3
+    - bls12-381
+    - pbkdf
+    - dirsp-proscript
+    - digestif
+  - name: Database
+    status: 🟠
+    description: |
+      n/a
+    packages:
+    - caqti
+    - pgocaml
+    - sqlite3
+    - lmdb
+    - irmin
+    - mysql
+    - postgresql
+    - aws-rds
+    - sequoia
+    - sqlgg
+    - petrol
+    - mariadb
+  - name: Email
+    status: 🟠
+    description: |
+      n/a
+    packages:
+    - tidy_email
+    - emile
+    - smtp
+    - letters
+    - mrmime
+    - sendmail
+    - facteur
+    - received
+    - email_message
+    - colombe
+  - name: En- & Decoding
+    status: 🟠
+    description: |
+      n/a
+    packages:
+    - tar
+    - imagelib
+    - base64
+    - multipart-form-data
+    - multipart_form
+    - biniou
+  - name: Web Frameworks
+    status: 🟠
+    description: |
+      n/a
+    packages:
+    - dream
+    - eliom
+    - opium
+    - sihl
+  - name: HTTP Clients
+    status: 🔴
+    description: |
+      n/a
+    packages:
+    - cohttp
+    - hyper
+    - ocurl
+    - http-lwt-client
+  - name: Internationalization
+    status: 🆘
+    description: |
+      n/a
+    packages:
+    - gettext
+  - name: Logging
+    status: 🟠
+    description: |
+      n/a
+    packages:
+    - logs
+    - dolog
+    - tracing
+    - catapult
+    - opentelemetry
+    - prometheus
+  - name: Lower Web-Stack
+    status: 🟡
+    description: |
+      n/a
+    packages:
+    - httpaf
+    - http
+    - tls
+    - websocket
+    - websocketaf
+    - gluten
+    - awa
+    - uri
+    - dns
+    - tcpip
+    - charrua
+    - mirage-nat
+    - arp
+    - ethernet
+    - paf
+    - mimic
+  - name: Serializers
+    status: 🟡
+    description: |
+      n/a
+    packages:
+    - atd
+    - yojson
```

Comment:
`jsonm` should be added too, but note that it doesn't produce standard conforming output if you serialize Float NaN/infinity values (the resulting json cannot be parsed), it require patching JSON parsers to accept the modified grammar.
`yojson` does at least produce standard conforming output, and the application can then handle NaN specifically (either as a null or string).However AFAICT yojson doesn't validate that its input is UTF-8 conforming.

My current stance is to use jsonm where validating input is important and use yojson where standard conforming output is important.

https://github.com/dbuenzli/jsonm/issues/12#issuecomment-1572025576

Reply:
I think this a wrong reading of `jsonm`'s api which [clearly states](https://erratique.ch/software/jsonm/doc/Jsonm/index.html#encoding_limitations) what's on the client to generate valid JSON. 

If fail to see how that's different from `yojson` which does not produces standard conforming output as it will happily output any byte string as an ill-coded json string; and I'm not even talking about the fact that it doesn't check the character encoding on input.

Reply:
I'd like to have at least one json library that takes and checks spec conformant JSON on input and products spec conformant JSON as output. Whether that is by patching 'jsonm' to implement the extra checks that 'yojson' does or patch 'yojson' to do the extra checks that 'jsonm' does. I'm currently attempting to fix it by doing the former :)

Reply:
I'd like to have at least one json library that takes and checks spec conformant JSON on input and products spec conformant JSON as output. Whether that is by patching 'jsonm' to implement the extra checks that 'yojson' does or patch 'yojson' to do the extra checks that 'jsonm' does. I'm currently attempting to fix it by doing the former :)

Reply:
There are various choices: raising an exception (so we never output something non-conformant), or serializing as `null`, or as a string. Either of these choices would allow the resulting JSON to be parsed and converted back at the application level to float if needed, it'd at least be conformant with the JSON grammar (it is what some other JSON libraries are doing, so there would even be some limited interoperability).
You are right that the choice should be left to the caller, however outputting invalid JSON is not a good default choice, because not even jsonm will be able to read back the output it just produced.

Jsonm already transforms strings on output (which is much more expensive in terms of performance), so why not floating point values?

Reply:
There are various choices: raising an exception (so we never output something non-conformant), or serializing as `null`, or as a string. Either of these choices would allow the resulting JSON to be parsed and converted back at the application level to float if needed, it'd at least be conformant with the JSON grammar (it is what some other JSON libraries are doing, so there would even be some limited interoperability).
You are right that the choice should be left to the caller, however outputting invalid JSON is not a good default choice, because not even jsonm will be able to read back the output it just produced.
OCaml has optional arguments so the library could implement one or more choices, and applications that really want  different semantic from the default can chose.

Jsonm already transforms strings on output (which is much more expensive in terms of performance), so why not floating point values?

Reply:
I'm adding `jsonm` to the list, thanks for the suggestion, it's a great library! I suggest continuing the discussion about the feature request on https://github.com/dbuenzli/jsonm/issues/22 though 😉